src/main/java/org/openlmis/buq/service/buq/BottomUpQuantificationService.java

```diff
@@ -312,4 +331,31 @@ private void validatePreparationParams(UUID facilityId, UUID programId,
     }
   }
 
+  /**
+   * Checks if a bottomUpQuantification of given period and facility exists.
+   *
+   * @param periodId UUID of period
+   * @param facilityId UUID of facility
+   * @return returns true if a buq of given period and facility exists. False otherwise.
+   */
+  public boolean existsByPeriodAndFacility(UUID periodId, UUID facilityId) {
+    Optional<BottomUpQuantification> bottomUpQuantification =
+            bottomUpQuantificationRepository
+                    .findByFacilityIdAndProcessingPeriodId(periodId, facilityId);
```

Comment:
Is this used only for checking if it exists? Then maybe we could try `existsByFacilityIdAndProcessingPeriodId`, might be supported:
`https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#appendix.query.method.subject`

Reply:
nice to know, I will check this out, thanks